### PR TITLE
project-proposal-<ss14432>-<at4524>

### PR DESCRIPTION
**Pull request by abitha-thankaraj**

### CODE REVIEW

File: ss14432-at4524/project.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+"peripheral" devices connect to a cloud. With the explosion of IoT devices and widespread availability 
+ of internet in the past decade, there has been a push to move computation to devices on the edge. While cloud computing
+ offers several advantages wih respect to elasticity and scalability, the distance between the edge devices and
+ the cloud gives rise to latency issues. This becomes a bottleneck in IoT applications
+ where results to computations are necessary in real-time (like in the case of disaster management
+and content delivery applications.) Service Level Agreements (SLAs) may also impose processing at 
+locations where the cloud provider does not have data centers. Latency and other limitations have 
+been discussed in this [paper](https://www.researchgate.net/profile/Rodolfo-Milito/publication/235409978_Fog_Computing_and_its_Role_in_the_Internet_of_Things/links/0deec531f19946228c000000/Fog-Computing-and-its-Role-in-the-Internet-of-Things.pdf).
+
+Resource management and task scheduling in Fog Computing have garnered interest especially since there has been a
```
Comment (vidit23) [inline ss14432-at4524/project.md:22]: What key differences do you see between cloud computing and fog computing in terms of task scheduling?

Comment (abitha-thankaraj) [reply ss14432-at4524/project.md:22]: 1. The applications in fog computing tend to be more latency sensitive. Latency becomes more important in fog scheduling 
2. Edge devices are limited in terms of resources such as memory and compute. 
3. Energy efficiency is also a deciding factor in edge computing schedulers when compared to cloud schedulers. Ie- the scheduling algo should not make the edge devices consume more power than necessary. 
4. You’re not always aware of the devices on the edge. They vary. But in cloud computing you tend to have a centralised place where you know which resources are available.

File: ss14432-at4524/project.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+* Initially, we will set up the simulator and create our edge architecture. The aim of this exercise is 
+to familiarise ourselves with the simulator and with edge architecture.
+
+* Next, we will then work on implementing a naive scheduler
+like FCFS, Round-Robin or SJF. This will allow us to understand custom schedulers in the 
+simulator and give use a baseline for comparison. 
+
+* We will then implement a more intelligent scheduler using our understanding of the constraints and scenarios of edge 
+and fog computing. (Note: There cannot be a one size fits all solution and
```
Comment (apanda) [inline ss14432-at4524/project.md]: How do you plan to generate these constraints?

What applications are you targetting?

**Review by apanda — COMMENTED** [review]
This proposal is currently missing a critical chunk: what scenarios are you considering scheduling, you at least need to provide an example application or scenario. There are reasonable tasks for which FCFS works perfectly well, and same with round-robin or SJF.

Please fix this ASAP, since I don't actually know what your project is aiming for at present.

File: ss14432-at4524/project.md (hunk 2 above)
Comment (SripranavSureshKumar) [reply ss14432-at4524/project.md]: The constraints can be set on the simulator for each of the devices and the servers. iFogSim provides the option to set constraints like latency for each device and power consumed by each device while busy and idle, etc. As for the values of these constraints we intend to set it specific to the devices we are modeling on our use case.

Comment (apanda) [reply ss14432-at4524/project.md]: What devices do you plan to model? How do you plan to get information about these devices?

Comment (SripranavSureshKumar) [reply ss14432-at4524/project.md]: We plan to model sensors and body connected IoT devices like ECG monitors, pulse oximeters, smart watches, etc. As for the information on the devices, we plan to identify datasets with the necessary details. We found another [paper](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0224934) which perfomed an analysis on latency of Healthcare IoT in the fog computing environment.

Comment (abitha-thankaraj) [reply ss14432-at4524/project.md]: Edge devices like - ECG, pulse oximeter, temperature sensor etc. 
Info about devices - To make it as close to real life as possible, we will use the same values of latency / network bandwidth as specified by papers that have collected data based on actual devices. One of the papers we plan to use is  - https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0224934

**Review by apanda — APPROVED** [review]
Thanks for adding the details. This is good for now, I look forward to seeing more information about the sensors and devices during the first checkin.

**Pull request merged**